internal/storage/v2/memory/tenant.go

```
@@ -75,3 +76,108 @@ func (t *Tenant) storeTraces(traceId pcommon.TraceID, resourceSpanSlice ptrace.R
 		t.ids[t.evict] = traceId
 	}
 }
+
+func (t *Tenant) findTraces(query tracestore.TraceQueryParams) iter.Seq2[[]ptrace.Traces, error] {
+	t.RLock()
+	defer t.RUnlock()
```

Please reread my comment.

---

So does that mean, firstly collect all the traces (slice of traces) and then iterating the slice to return the traces in iterator?

---

Also what should we do if search depth of query is negative? Should we return error or assume no search depth?

---

yes, we can return an error for `SearchDepth<=0`